pkg/prometheus/promcfg_test.go

```diff
@@ -6444,3 +6444,500 @@ func TestTracingConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestServiceMonitorScrapeClass(t *testing.T) {
+	// test table for scrape class
+	testCases := []struct {
+		name        string
+		scrapeClass *monitoringv1.ScrapeClass
+		golden      string
+		expectedErr bool
```

Comment:
expectedErr is false in all test cases... do we really need it? Maybe you forgot a test case where an error was expected? 😛

Reply:
Ups!
Removed 😅